### ISSUE TITLE
Build a Streaming Aggregation Processor For Service Map

### ISSUE THREAD

**Issue by mahadzaryab1** [jaegertracing/jaeger#6153]
### Requirement

Implement a processor in the OTEL pipeline that takes as input a stream of traces and constructs a dependency graph between services. 

### Problem

N/A

**Comment by mahadzaryab1** [comment]
@yurishkuro in this first step, do we just want to compute the dependencies in an apache beam pipeline?

**Comment by yurishkuro** [comment]
Yes but also implement that as an OTEL Processor. You can probably copy the existing processor from contrib (mentioned in the comments above) as a baseline.

**Comment by mahadzaryab1** [comment]
@yurishkuro Are you referring the tailsamplingprocessor? https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/processor/tailsamplingprocessor

**Comment by mahadzaryab1** [comment]
@yurishkuro Another question - looks like OTEL has a [groupbytraceprocessor](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/processor/groupbytraceprocessor). Is there any way we could leverage this or do we need to reuse this logic inside the processor that we write?

**Comment by yurishkuro** [comment]
No, I was referring to [ServiceGraphConnector](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/connector/servicegraphconnector/README.md)

[groupbytraceprocessor](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/processor/groupbytraceprocessor) is not needed for memory/badger, but may be needed for distributed storage. It somewhat depends on how Apache Beam implements the streams - if we have to shard spans by trace ID manually then something from OTEL will be needed (it's the same problem as with tail sampling processor, which needs full traces in one collector).
